index.bs

```diff
@@ -2542,6 +2642,63 @@ To <dfn>obtain and deliver a debug report on source registration</dfn> given a
+    1. [=set/Remove=] |source| from the [=attribution source cache=].
+
+To <dfn>remove sources with lower attribution scope limit from cache if necessary</dfn> given an [=attribution source=] |pendingSource|:
+1. If |pendingSource|'s [=attribution source/attribution scope limit=] is null, return.
+1. Let |matchingSources| be the result of running [=find sources with common destinations and reporting origin=] with |pendingSource|.
+1. [=list/iterate|For each=] |source| of |matchingSources|:
+    1. If |source|'s [=attribution source/attribution scope limit=] is null or greater than or equal to |pendingSource|'s [=attribution source/attribution scope limit=], [=iteration/continue=].
+    1. [=Remove fake event reports for source=] with |source| and |pendingSource|'s [=attribution source/source time=].
+    1. [=set/Remove=] |source| from the [=attribution source cache=].
+
+To <dfn>remove attribution scopes for sources with destination</dfn> given a positive 32-bit integer |attributionScopeLimit|, a [=site=] |destination|, and a [=suitable origin=] |reportingOrigin|:
+1. Let |attributionScopeRecords| be a new [=list=].
+1. [=set/iterate|For each=] |source| of the [=attribution source cache=]:
+    1. If |source|'s [=attribution source/reporting origin=] and |pendingSource|'s [=attribution source/reporting origin=] are not [=same origin=], [=iteration/continue=].
+    1. If |source|'s [=attribution source/attribution destinations=] does not [=set/contains|contain=] |destination|, [=iteration/continue=].
+    1. If |source|'s [=attribution source/attribution scopes=] is null or [=set/is empty|empty=], [=iteration/continue=].
+    1. [=set/iterate|For each=] |attributionScope| in |source|'s [=attribution source/attribution scopes=]:
+        1. [=list/Append=] the [=tuple=] (|attributionScope|, |source|'s [=attribution source/source time=], |source|'s [=attribution source/source identifier=]) to |attributionScopeRecords|.
+1. [=list/sort in ascending order|Sort=] |attributionScopeRecords| in ascending order with |a| being less than |b| if any of the following are true:
+      * |a|[1] is greater than |b|[1].
+      * |a|[1] is equal to |b|[1] and |a|[0] is greater than |b|[0].
+1. Let |selectedAttributionScopes| be a new [=set=].
+1. Let |sourceIdsToRemove| be a new [=set=].
+1. [=list/iterate|For each=] [=tuple=] |record| of |attributionScopeRecords|:
+    1. If |selectedAttributionScopes|'s [=set/size=] is less than |attributionScopeLimit|, [=set/append=] |record|[0] to |selectedAttributionScopes|.
+    1. If |selectedAttributionScopes| does not [=set/contain=] |record|[0], [=set/append=] |record|[2] to |sourceIdsToRemove|.
+1. [=set/iterate|For each=] |source| of the [=attribution source cache=]:
+    1. Otherwise, if |sourceIdsToRemove| [=set/contains=] |source|'s [=attribution source/source identifier=], set |source|'s [=attribution source/attribution scopes=] to null.
```

Comment:
I don't think the `Otherwise, ` prefix here is right, as there's no preceding condition.

Reply:
Done. Otherwise is meant for last if block :)